File: src/main/java/com/gmail/borlandlp/minigamesdtools/creator/DataProvider.java
package com.gmail.borlandlp.minigamesdtools.creator;

import java.util.HashSet;
import java.util.Hashtable;
import java.util.Map;
import java.util.Set;

public class DataProvider implements AbstractDataProvider {
    private Map<String, Object> data = new Hashtable<>();

    @Override
    public boolean contains(String key) {
        return this.data.containsKey(key);
    }

    @Override
    public Object get(String key) {
        return this.data.get(key);
    }

    @Override
    public void set(String key, Object value) {
        this.data.put(key, value);
    }

    @Override
    public void remove(String key) {
        this.data.remove(key);
    }

    @Override
    public Set<String> getKeys() {
        return new HashSet<>(this.data.keySet());
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/creator/DataProvider.kt
package com.gmail.borlandlp.minigamesdtools.creator

import java.util.*

open class DataProvider : AbstractDataProvider {
    private val data: MutableMap<String, Any> =
        Hashtable()

    override fun contains(key: String): Boolean {
        return data.containsKey(key)
    }

    override fun get(key: String): Any {
        return data[key]!!
    }

    override fun set(key: String, value: Any) {
        data[key] = value
    }

    override fun remove(key: String) {
        data.remove(key)
    }

    override val keys: Set<String>
        get() = HashSet(data.keys)
}